cmd/create.go

```diff
@@ -20,114 +22,177 @@ func init() {
 	root.AddCommand(NewCreateCmd(newCreateClient))
 }
 
+// createClientFn is a factory function which returns a Client suitable for
+// use with the Create command.
+type createClientFn func(createConfig) (*fn.Client, error)
+
 // newCreateClient returns an instance of fn.Client for the "Create" command.
 // The createClientFn is a client factory which creates a new Client for use by
 // the create command during normal execution (see tests for alternative client
 // factories which return clients with various mocks).
-func newCreateClient(cfg createConfig) *fn.Client {
-	return fn.New(
-		fn.WithRepositories(cfg.Repositories),
-		fn.WithRepository(cfg.Repository),
-		fn.WithVerbose(cfg.Verbose))
+func newCreateClient(cfg createConfig) (*fn.Client, error) {
+	client := fn.New(
+		fn.WithRepositories(cfg.Repositories), // path to repositories in disk
+		fn.WithRepository(cfg.Repository),     // URI of repository override
+		fn.WithVerbose(cfg.Verbose))           // verbose logging
+	return client, nil
 }
 
-// createClientFn is a factory function which returns a Client suitable for
-// use with the Create command.
-type createClientFn func(createConfig) *fn.Client
-
 // NewCreateCmd creates a create command using the given client creator.
 func NewCreateCmd(clientFn createClientFn) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "create [PATH]",
-		Short: "Create a function project",
-		Long: `Create a function project
-
-Creates a new function project in PATH, or in the current directory if no PATH is given. 
-The name of the project is determined by the directory name the project is created in.
-`,
-		Example: `
-# Create a Node.js function project in the current directory, choosing the
-# directory name as the project's name.
-kn func create
-
-# Create a Quarkus function project in the directory "sample-service". 
-# The directory will be created in the local directory if non-existent and 
-# the project is called "sample-service"
-kn func create --runtime quarkus sample-service
-
-# Create a function project that uses a CloudEvent based function signature
-kn func create --template events sample-events
-	`,
+		Use:   "create",
+		Short: "Create a Function Project",
+		Long: `
+NAME
+	{{.Prefix}}func create - Create a Function project.
+
+SYNOPSIS
+	func create [-l|--language] [-t|--template] [-r|--repository]
+	             [-c|--confirm]  [-v|--verbose] [path]
+
+DESCRIPTION
+	Creates a new Function project.
+
+	  $ {{.Prefix}}func create -l go -t http
+
+	Creates a Function in the current directory '.' which is written in the
+	language/runtime 'Go' and handles HTTP events.
```

Comment:
Really trying not to bikeshed here, but one comment from user feedback was that it would be nice if the most likely, or at least most illustrative, example command was shown right up top in the help text.

I presume we'll be steadily improving these docs over time, so this is just a gentle suggestion in an attempt at an iterative improvement.